Add documentation on refactoring view helpers

gsteel opened the pull request:
Also makes some improvements to the migration guide

gsteel committed:
Add documentation on refactoring view helpers

Also makes some improvements to the migration guide

Signed-off-by: George Steel <george@net-glue.co.uk>

froschdesign commented on docs/book/v3/migration/refactoring-view-helpers.md:
```diff
@@ -0,0 +1,131 @@
+            PluginWithDependencies::class => PluginFactory::class,
+        ],
+        'aliases' => [
+            'myPlugin' => PluginWithDependencies::class,
```
Perhaps a note could be added to explain that the alias is required for usage in the view script/template.

gsteel committed:
Add note about aliases

Signed-off-by: George Steel <george@net-glue.co.uk>

froschdesign commented on docs/book/v3/migration/refactoring-view-helpers.md:
```diff
@@ -0,0 +1,135 @@
+> The key defined under `aliases` is how you _access_ the helper from within view scripts.
+> See the [Advanced Usage chapter](../helpers/advanced-usage.md#writing--registering-custom-helpers) for further information.
+
+The refactored plugin is actually compatible with both `laminas-view` version 2 _and_ version 3.
+It has always been possible to use dependency injection with `laminas-view`, therefore refactoring of plugins can happen prior to upgrading.
```
Great detail and I think a link from the chapter "[Preparing for Version 3](https://docs.laminas.dev/laminas-view/v2/migration/preparing-for-v3/)" in documentation of version 2 would be good.

gsteel committed:
Update preparation guide.

- Corrects usage examples of resolvers
- Updates list of removed helpers
- Links back to new helper refactoring guide

Signed-off-by: George Steel <george@net-glue.co.uk>

froschdesign commented:
@gsteel 
Thank you! 👍🏻
